resources/brave-youtube-quality.js

```diff
@@ -0,0 +1,56 @@
+/// brave-youtube-quality.js
+(function() {
+    const IS_MOBILE_YOUTUBE = window.location.hostname == 'm.youtube.com';
+    const IS_ANDROID = window.navigator.userAgent.indexOf('Android') > -1;

```

Comment:
why don't you decide whether to inject the script or not in native (java or cpp)?

Reply:
@stoletheminerals it's not possible if we're using this via adblock scriptlet injections; those can be selectively injected per-domain, but not according to any other criteria. If you want to move checks to native code it'll have to be a completely separate injection mechanism.

Reply:
@antonok-edm thanks, yeah that makes sense. I'm just not sure why we decided to move this code to adblock scriptlets. To me it's weird to inject first and then check if we injected it on Android, when we could decide before the injection, but I understand that we can't do it in the current implementation. Anyway, it's not a security concern.